lib/web_ui/dev/steps/run_tests_step.dart

```diff
@@ -164,6 +175,54 @@ class RunTestsStep implements PipelineStep {
     }
     return message.toString();
   }
+
+  Future<SkiaGoldClient?> _createSkiaClient() async {
+    const FileSystem fs = LocalFileSystem();
+    final Directory goldensRoot =
+        fs.directory(environment.webUiSkiaGoldDirectory.path);
+    final SkiaGoldClient skiaClient =
+        SkiaGoldClient(goldensRoot, browserName: browserName);
+
+    if (!await _checkSkiaClient(skiaClient)) {
+      print('WARNING: Unable to use Skia Client in this environment.');
```

Comment:
I think if there is an intention to perform golden comparisons then this should be a hard failure. Otherwise this check can lead to false positives on the CI. This happened before several times.

Reply:
Agreed!

Reply:
Before making this a hard failure, there may be some cases where a hard failure is undesirable. In the framework there are cases where we can't use the skia client. Not sure if these cases also apply here:
1. **Local framework testing** - Since the skia client requires authentication, local testing cannot use the skia client. In this case for the framework, we use the `FlutterLocalFileComparator`, which manually retrieves the image for the appropriate local environment using Gold public API.
2. **Other repositories** - Other repositories cannot talk to Gold, since Gold associates images with changes for a single repository. So for the framework Gold instance, other repos that check out and run framework tests like Dart, Skia and the Engine, they would not be able to run these tests with the client. These cases are handled in the framework with the `FlutterSkippingFileComparator`, and the golden file tests are run when those repos roll _into_ the framework.

Reply:
@Piinks absolutely! We are not going to make it always a hard failure. There'll be a flag passed in command line (`--require-skia-gold`) that controls whether we should fail or not.

For example, in Windows web tests, we don't take any screenshots and we don't expect to do any comparison, so we shouldn't require Skia Gold to be available there.

But in Linux Chrome tests, we want to make sure that Skia Gold is working and screenshots are being uploaded.

Reply:
Gotcha! Thanks for explaining! :)